pkg/controllers/workloadrebalancer/workloadrebalancer_controller.go

```diff
@@ -84,18 +92,34 @@ func (c *RebalancerController) Reconcile(ctx context.Context, req controllerrunt
 	}
 
 	// 2. get and update referenced binding to trigger a rescheduling
-	newStatus, successNum, retryNum := c.doWorkloadRebalance(ctx, rebalancer)
+	result := c.doWorkloadRebalance(ctx, rebalancer)
 
-	// 3. update status of WorkloadRebalancer
-	if err := c.updateWorkloadRebalancerStatus(ctx, rebalancer, newStatus); err != nil {
-		return controllerruntime.Result{}, err
+	// 3. update status of WorkloadRebalancer if statusChanged
+	if result.statusChanged {
+		if err := c.updateWorkloadRebalancerStatus(ctx, rebalancer, result.newStatus); err != nil {
+			return controllerruntime.Result{}, err
+		}
 	}
+
 	klog.Infof("Finish handling WorkloadRebalancer (%s), %d/%d resource success in all, while %d resource need retry",
```

Comment:
Is this line of log really effective in locating the problem or is the log level too high? Logs about the processing failure of each workload are generated.